torchvision/prototype/datasets/datapipes.py

```diff
@@ -0,0 +1,29 @@
+from typing import Tuple, IO, Iterator, Union, cast
+
+import torch
+from torchdata.datapipes.iter import IterDataPipe
+
+__all__ = ["FloReader"]
+
+
+class FloReader(IterDataPipe[torch.Tensor]):
+    def __init__(self, datapipe: IterDataPipe[Tuple[str, IO]]) -> None:
+        self.datapipe = datapipe
+
+    def _read_data(self, file: IO, *, dtype: torch.dtype, count: int) -> torch.Tensor:
```

Comment:
IMHO this helper is not really necessary, same for `_read_scalar`. The original code is fairly short and having 1 or 2 levels of indirection when reading it makes it harder to follow

Reply:
I can get onboard with the `_read_scalar`, but I feel like `_read_data` is quite useful. It isn't a one-liner and especially with the endian handling will get even more logic.